codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/AWSSigningMiddleware.kt

```diff
@@ -57,12 +65,22 @@ open class AWSSigningMiddleware(
     }
 
     private fun middlewareParamsString(op: OperationShape): String {
-        paramsCallback?.let {
-            return it(op)
-        } ?: run {
-            val hasUnsignedPayload = op.hasTrait<UnsignedPayloadTrait>()
-            return "unsignedBody: $hasUnsignedPayload"
-        }
+        val serviceIsS3 = service.sdkId.lowercase(Locale.US) == "s3"
+        val serviceIsGlacier = service.sdkId.lowercase(Locale.US) == "glacier"
+        val useUnsignedPayload = op.hasTrait<UnsignedPayloadTrait>() || params.forceUnsignedBody
+        val useSignedBodyHeader = (serviceIsS3 || serviceIsGlacier) && !useUnsignedPayload
+
+        // Create param strings for each setting, or null for default param
+        val params = listOf(
+            "signatureType: .requestQueryParams".takeIf { params.useSignatureTypeQueryString },
+            "useDoubleURIEncode: false".takeIf { serviceIsS3 },
+            "shouldNormalizeURIPath: false".takeIf { serviceIsS3 },
+            "expiration: expiration".takeIf { params.useExpiration },
+            "signedBodyHeader: .contentSha256".takeIf { useSignedBodyHeader },
+            "unsignedBody: " + ("true".takeIf { useUnsignedPayload } ?: "false")
+        )
```

Comment:
Can we add properties to `AWSSigningParams` for all the values here? Such as `useDoubleURIEncode` `shouldNormalizeURIPath`, and `useSignedBodyHeader`. 
Then can we add an initializer to `AWSSigningParams` that takes a service as an input and performs the service specific logic to configure the parameters?

This makes this more testable as this chunk of code simply converts `AWSSigningParams` to a string. 
And the various initializers on `AWSSigningParams` are responsible for the setting the properties correctly according to the service. This makes things a bit easier to change and test in the future as we add/remove service specific handlings.

Reply:
Done.  `AWSSigningParams` absorbed the logic out of `AWSSigningMiddleware`, and now has its own tests in `AWSSigningMiddlewareTests`.

Reply:
I have now additionally broke the logic away from `AWSSigningParams` into `SigV4Configurator` which generates the param string for use with the Swift middleware.

`SigV4Configurator` has a convenience initializer that takes a set of params & transforms those into which fields should be set in the Swift param string.  This improves testability by exposing the inputs to the param string directly.